### DISCUSSION TITLE
Migrating from v2 to v3 - how to substitute 'relate' tag

### DISCUSSION THREAD

**Discussion by cmorierroy** [statamic/cms#5458] [Troubleshooting] [answered]
Hi, I'm migrating a site from v2 to v3 and most things work so far. 
I am running into problems with the relate tag - I know it's a breaking change, but the augmentation part of the doc confuses me.

This is what the original code looked like, country being an entry field on location entries.

    {{ collection:location scope="loc" }}
        {{ relate:country }}
            {{if country=="e69a915e-c26f-4f0a-a6ae-762ef6ec80c0"}}
                <a onclick="locationSelected('{{ loc:slug }}')" href="/{{ loc:slug }}">{{ loc:title }}</a><br>
            {{/if}}
        {{ /relate:country }}
    {{ /collection:location }}

What I have now (doesn't show anything, but doesn't throw an error):

    {{ collection:location scope="loc" }}
        {{ if loc:country=="canada" }}
            <a onclick="locationSelected('{{ loc:slug }}')" href="/{{ loc:slug }}">{{ loc:title }}</a><br>
        {{/if}}
    {{ /collection:location }}

I'm aware that my condition might not be correct, but it's something like that I'm hoping to achieve, or {{ loc:country:name }}.
Within {{ collection:location scope="loc" }}, {{ loc:country }} doesn't give me anything, as well as {{ country }}.
However, things like {{ loc:name }} work.

Any ideas for how I would fix this? I'm just trying to access the country entry on each location entry.

**Comment by jasonvarga** [comment]
Does your `country` field have `max_items: 1` on it? 
If you intend for it to be a single entry, you'll need that.

**Reply by cmorierroy** [reply]
Yes it does.

**Reply by jasonvarga** [reply]
Did your original code not work?

**Reply by cmorierroy** [reply]
The original code throws this:

Facade\Ignition\Exceptions\ViewException
addcslashes(): Argument #1 ($string) must be of type string, Statamic\Query\OrderedQueryBuilder given

I figure it's due to using the relate keyword

**Reply by jasonvarga** [reply]
Oh! That looks like something I fixed recently. You didn't mention you were trying to use the 3.3 beta.

https://github.com/statamic/cms/pull/5461

I haven't tagged it yet, it'll be in the next release.

**Reply by cmorierroy** [reply]
That's gotta be it.  Sorry, I did not notice I was on beta, but yes, I'm definitely on 3.3 now that I'm checking.

**Comment by cmorierroy** [comment, marked as answer]
Well... I deleted the country fieldset from the location collection and re-added it.
Now things work! Maybe something migrated weirdly? 
Things work as in, this syntax works:

    {{ collection:location scope="loc" }}
        {{if country:slug == "canada"}}
            <a onclick="locationSelected('{{ loc:slug }}')" href="/{{ loc:slug }}">{{ loc:title }}</a><br>
        {{/if}}
    {{ /collection:location }}

**Reply by jackmcdade** [reply]
Do you have a diff of the before/after?

**Reply by cmorierroy** [reply]
![image](https://user-images.githubusercontent.com/16982565/157907123-d364da8d-57cc-4f3a-8f29-f49aa0908587.png)

**Reply by jackmcdade** [reply]
i wonder if it's the `mode: tags` vs `mode: default`?

**Reply by jasonvarga** [reply]
My guess is `max_items: '1'` vs. `max_items: 1`
